vignettes/applications.Rmd

```diff
@@ -247,3 +247,167 @@ print(gg.out)
+  win( 148, 171),
+  win( 354, 361),
+  win(1059,1065))
+mb.fac <- 1e6
+wfac <- function(x){
+  factor(x, c("6.5-7.1", "14.8-17.1", "35.4-36.1", "105.9-106.5"))
+}
+windows[, window := wfac(sprintf(
+  "%.1f-%.1f", windowStart/mb.fac, windowEnd/mb.fac))]
+setkey(windows, windowStart, windowEnd)
+f <- function(dt, key.vec){
+  setkeyv(dt, key.vec)
+  dt
+}
+profile614chr2$probes[, pos0 := position]
+over.list <- list(
+  changes=f(some.change, c("change", "segStart")),
+  segments=f(some.segs, c("segStart", "segEnd")),
+  labels=f(profile614chr2$labels, c("labelStart", "labelEnd")),
+  errors=f(err.dt, c("labelStart", "labelEnd")),
+  probes=f(profile614chr2$probes, c("position", "pos0")))
+join.list <- lapply(over.list, foverlaps, windows, nomatch=0L)
+
+show.err <- join.list$errors[, list(
+  fp=sum(fp),
+  fn=sum(fn),
+  errors=sum(fp+fn)
+  ), by=list(graph.name, n.segments)]
+
+library(ggplot2)
+br <- c(6.5,7.0,seq(15,17,by=0.5),35.5,36,106,106.5)
+names(br) <- as.character(br)
+gg.out <- ggplot()+
+  theme_bw()+
+  theme(panel.margin=grid::unit(0, "lines"))+
+  coord_cartesian(expand=FALSE)+
+  facet_grid(
+    graph.name + n.segments ~ window,
+    ##labeller=label_both,
+    scales="free",space="free")+
+  penaltyLearning::geom_tallrect(aes(
+    xmin=labelStart/mb.fac, xmax=labelEnd/mb.fac, fill=annotation),
+    color="grey",
+    data=join.list$labels)+
+  scale_fill_manual(
+    "label",
+    values=penaltyLearning::change.colors)+
+  penaltyLearning::geom_tallrect(aes(
+    xmin=labelStart/mb.fac, xmax=labelEnd/mb.fac, linetype=status),
+    fill=NA,
+    size=1,
+    color="black",
+    data=join.list$errors)+
+  scale_linetype_manual(
+    "error type",
+    limits=c("correct", 
+             "false negative",
+             "false positive"),
+    values=c(correct=0,
+             "false negative"=3,
+             "false positive"=1))+
+  geom_point(aes(
+    position/mb.fac, logratio),
+    shape=1,
+    color="grey50",
+    data=join.list$probes)+
+  scale_y_continuous(
+    "Logratio (Measure of DNA copy number)",
+    breaks=seq(-2, 2, by=2)
+  )+
+  scale_x_continuous("Position on chr2 (mega bases)", breaks=br)+
+  scale_color_manual(values=c(
+    std="green",
+    absSup="deepskyblue"))+
+  geom_segment(aes(
+    ifelse(segStart < windowStart, windowStart, segStart)/mb.fac, mean,
+    color=graph.name,
+    xend=ifelse(windowEnd < segEnd, windowEnd, segEnd)/mb.fac, yend=mean),
+    data=join.list$segments,
+    size=1)+
+  geom_vline(aes(
+    xintercept=change/mb.fac, color=graph.name),
+    linetype="dashed",
+    size=0.75,
+    data=join.list$changes)+
+  geom_text(aes(
+    14.8, -3, label=sprintf(
+      " %d label error%s for %d segment %s model",
+      errors, ifelse(errors==1, "", "s"), n.segments, graph.name)),
+    hjust=0,
+    vjust=0,
+    data=data.table(show.err, window=wfac("14.8-17.1")))
+##png("figure-relevant-changes-copy-number.png", 8, 4, units="in", res=300, type="cairo")
+print(gg.out)
```

Comment:
I have also added vignette R code that shows how to make the absSup model figure for the DNA copy number data set.